code/modules/projectiles/guns/energy/hos.dm

```diff
@@ -0,0 +1,273 @@
+
+	var/x_offset = 5
+	var/current = 0
+	for(var/B in stored_ammo)
+		var/obj/item/ammo_casing/mws_batt/batt = B
+		var/image/cap = image(icon, icon_state = "[initial(icon_state)]_cap")
+		cap.color = batt.type_color
+		cap.pixel_x = current * x_offset //Caps don't need a pixel_y offset
+		add_overlay(cap)
+		if(batt.cell.charge > 0)
+			var/ratio = CEILING(clamp(batt.cell.charge / batt.cell.maxcharge, 0, 1) * 4, 1) //4 is how many lights we have a sprite for
+			var/image/charge = image(icon, icon_state = "[initial(icon_state)]_charge-[ratio]")
+			charge.color = "#29EAF4" //Could use battery color but eh.
+			charge.pixel_x = current * x_offset
+			add_overlay(charge)
+
+		current++ //Increment for offsets
+
+// The Casing //
+/obj/item/ammo_casing/mws_batt
+	name = "\'MWS\' microbattery - UNKNOWN"
+	desc = "A miniature battery for an energy weapon."
+	icon = 'icons/obj/ammo.dmi'
+	icon_state = "mws_batt"
+	slot_flags = SLOT_BELT | SLOT_EARS
+	throwforce = 1
+
+	caliber = "mws"
+	var/type_color = null
+	var/type_name = null
+
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell{charge = 600; maxcharge = 600}
+
+	var/e_cost = 100
+	projectile_type = /obj/item/projectile/beam
+
+/obj/item/ammo_casing/mws_batt/Initialize()
+	. = ..()
+	pixel_x = rand(-10, 10)
+	pixel_y = rand(-10, 10)
+	cell = new cell_type(src)
+	cell.give(cell.maxcharge)
+	update_overlays()
+
+/obj/item/ammo_casing/mws_batt/update_overlays()
+	.=..()
+	cut_overlays()
+
+	var/image/ends = image(icon, icon_state = "[initial(icon_state)]_ends")
+	ends.color = type_color
+	add_overlay(ends)
+
+/obj/item/ammo_casing/mws_batt/get_cell()
+	return cell
+
+/obj/item/ammo_casing/mws_batt/proc/chargeshot()
+	if(cell.charge >= e_cost)
+		cell.use(e_cost)
+		newshot()
+		return
+	else
+		return
+
+// Specific batteries //
+/obj/item/ammo_casing/mws_batt/lethal
+	name = "'MWS' microbattery - LETHAL"
+	type_color = "#bf3d3d"
+	type_name = "<span style='color:#bf3d3d;font-weight:bold;'>LETHAL</span>"
+	projectile_type = /obj/item/projectile/beam
+
+/obj/item/ammo_casing/mws_batt/stun
+	name = "'MWS' microbattery - STUN"
+	type_color = "#0f81bc"
+	type_name = "<span style='color:#0f81bc;font-weight:bold;'>STUN</span>"
+	projectile_type = /obj/item/projectile/beam/disabler
+
+/obj/item/ammo_casing/mws_batt/net
+	name = "'MWS' microbattery - NET"
+	type_color = "#43f136"
+	type_name = "<span style='color:#43d136;font-weight:bold;'>NET</span>"
+	projectile_type = /obj/item/projectile/energy/net
+
+/obj/item/ammo_casing/mws_batt/xray
+	name = "'MWS' microbattery - XRAY"
+	type_color = "#32c025"
+	type_name = "<span style='color:#32c025;font-weight:bold;'>XRAY</span>"
+	projectile_type = /obj/item/projectile/beam/xray
+
+
+/obj/item/ammo_casing/mws_batt/ion
+	name = "'MWS' microbattery - ION"
+	type_color = "#d084d6"
+	type_name = "<span style='color:#d084d6;font-weight:bold;'>ION</span>"
+	projectile_type = /obj/item/projectile/ion
+
+
+/obj/item/choice_beacon/hosgun
```

Comment:
Move this to `code/game/objects/items/miscellaneous.dm`, where other beacons are.